src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,8 +1,37 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
+import { useState } from "react";
+
+type TodoFilterProps = {
+  onSearchChange: (searchTerm: string) => void;
+  onStatusChange: (status: string) => void;
+};
+
+export const TodoFilter: React.FC<TodoFilterProps> = ({onSearchChange, onStatusChange}) => {
+  const [selectedStatus, setSelectedStatus] = useState<string>("all");
+  const [searchTerm, setSearchTerm] = useState<string>("");
```

Comment:
why do you need two the same states like in the App, it's redundant, just pass the values of those states as props here